test/metric/dimension/custom_provider.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package dimension
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"log"
+)
+
+type CustomDimensionProvider struct {
+	Provider
+}
+
+var _ IProvider = (*CustomDimensionProvider)(nil)
+
+func (p *CustomDimensionProvider) IsApplicable() bool {
+	return true
+}
+
+func (p *CustomDimensionProvider) GetDimension(instruction Instruction) types.Dimension {
+	log.Printf("CustomDimProvider")
```

Comment:
were these temp prints or did you mean to keep these?

Reply:
yeah that's not too useful I guess